### PR TITLE
Add more features to span and unit tests

### PR DESCRIPTION
**Pull request by pelszkow**

### CODE REVIEW

**Review by tomdol — APPROVED** [review]

**Review by jdanieck — APPROVED** [review]

File: ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp [hunk 1]
```diff
@@ -92,6 +92,34 @@ namespace ngraph
                     return {m_data + offset, std::min(size, m_size - offset)};
                 }
 
+                Span& drop_front(std::size_t number_of_elements)
```
Comment (blesniewski) [inline ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:95]: What do you think of setting the default `number_of_elements` value to 1?

Comment (pelszkow) [reply ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:95]: I prefer to be explicit during the call so I won't default here.

**Review by tomdol — APPROVED** [review]

**Comment by jdanieck** [comment]
@pelszkow you requested re-review but because of your force-push I don't see what was changed, so please explain.

**Comment by pelszkow** [comment]
New implementation in span was completely changed. Some unit test survive unchanged but I can't see reason to store history.

File: ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp [hunk 2]
```diff
@@ -92,6 +92,34 @@ namespace ngraph
...
+                {
+                    if (number_of_elements < m_size)
+                    {
+                        m_size -= number_of_elements;
+                    }
+                    else
+                    {
+                        m_size = 0;
+                    }
+
```
Comment (lazarevevgeny) [inline ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:119]: nit: remove empty line

File: ngraph/test/span.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+#include <algorithm>
+#include <array>
+#include <vector>
+
+#include "ngraph/runtime/reference/utils/span.hpp"
+
+TEST(span_util, create_from_vector)
+{
+    using namespace ngraph::runtime::reference;
```
Comment (lazarevevgeny) [inline ngraph/test/span.cpp:15]: move this line out of all `TEST`.

File: ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp [hunk 4]
```diff
@@ -92,6 +92,34 @@ namespace ngraph
...
+                        m_size -= number_of_elements;
+                    }
+                    else
+                    {
+                        m_size = 0;
+                    }
+                    return *this;
+                }
+
+                Span& drop_back(std::size_t number_of_elements)
```
Comment (lazarevevgeny) [inline ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:109]: Add comment describing the function behaviour.

File: ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp (hunk 1 above)
Comment (lazarevevgeny) [inline ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:95]: Add comment describing the function behaviour.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp (hunk 2 above)
Comment (pelszkow) [reply ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:119]: done

File: ngraph/test/span.cpp (hunk 3 above)
Comment (pelszkow) [reply ngraph/test/span.cpp:15]: done

File: ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp (hunk 1 above)
Comment (pelszkow) [reply ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:95]: done

File: ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp (hunk 4 above)
Comment (pelszkow) [reply ngraph/core/reference/include/ngraph/runtime/reference/utils/span.hpp:109]: done

**Review by lazarevevgeny — APPROVED** [review]

**Review by jdanieck — APPROVED** [review]

**Comment by jdanieck** [comment]
@lazarevevgeny can we merge? This PR is needed for #503.

**Pull request merged**